File: be-engine-v1/be-flow-activiti/src/main/java/com/cmcorg20230301/be/engine/flow/activiti/model/enums/SysActivitiParamItemTypeEnum.java
package com.cmcorg20230301.be.engine.flow.activiti.model.enums;

import com.cmcorg20230301.be.engine.flow.activiti.model.interfaces.ISysActivitiParamItemType;
import com.cmcorg20230301.be.engine.flow.activiti.util.SysActivitiUtil;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public enum SysActivitiParamItemTypeEnum implements ISysActivitiParamItemType {

    TEXT(101), // 文字（默认）

    IMAGE(201), // 图片：url

    FUNCTION_CALL_SIMPLE(301), // 函数调用

    FUNCTION_CALL_CUSTOM(401), // 函数调用-高级

    URL(501), // 网络链接

    ;

    private final int code;

    static {

        for (SysActivitiParamItemTypeEnum item : SysActivitiParamItemTypeEnum.values()) {

            SysActivitiUtil.PARAM_ITEM_TYPE_MAP.put(item.getCode(), item);

        }

    }

}

File: fe-antd-v1/src/model/enum/SysActivitiParamItemTypeEnum.ts
import {IEnum} from "@/model/enum/CommonEnum";
import {ProSchemaValueEnumType} from "@ant-design/pro-components";

export interface ISysActivitiParamItemTypeEnum {

    TEXT: IEnum<number>,

    IMAGE: IEnum<number>,

    URL: IEnum<number>,

}

// 流程图参数类型，枚举类
export const SysActivitiParamItemTypeEnum: ISysActivitiParamItemTypeEnum = {

    TEXT: {
        code: 101,
        name: '文字',
    },

    IMAGE: {
        code: 201,
        name: '图片路径',
    },

    URL: {
        code: 501,
        name: '网络链接',
    },

}

export const SysActivitiParamItemTypeEnumDict = new Map<number, ProSchemaValueEnumType>();

Object.keys(SysActivitiParamItemTypeEnum).forEach(key => {

    const item = SysActivitiParamItemTypeEnum[key] as IEnum<number>;

    SysActivitiParamItemTypeEnumDict.set(item.code!, {text: item.name})

})
